lib/Elastica/Reindex.php

```diff
@@ -0,0 +1,139 @@
+<?php
+namespace Elastica;
+
+use Elastica\Query\AbstractQuery;
+
+class Reindex
+{
+    const VERSION_TYPE = 'version_type';
+    const VERSION_TYPE_INTERNAL = 'internal';
+    const VERSION_TYPE_EXTERNAL = 'external';
+    const OPERATION_TYPE = 'op_type';
+    const OPERATION_TYPE_CREATE = 'create';
+    const CONFLICTS = 'conflicts';
+    const CONFLICTS_PROCEED = 'proceed';
+    const TYPE = 'type';
+    const SIZE = 'size';
+    const QUERY = 'query';
+
+    /**
+     * @var Index
+     */
+    private $oldIndex;
+
+    /**
+     * @var Index
+     */
+    private $newIndex;
```

Comment:
Please make these all protected and not private. In case someone wants to extend Reindex in his code, he should be able to.